testutil/validatormock/validatormock.go

```diff
@@ -164,7 +188,12 @@ func ProposeBlock(ctx context.Context, eth2Cl Eth2Provider, signFunc SignFunc, s
 		}
 
 		// Get Unsigned beacon block with given randao and slot
-		block, err = eth2Cl.BeaconBlockProposal(ctx, slot, randao, nil)
+		if addr == "" {
+			// For testing purposes assuming beaconmock is used in place of validatorapi
+			block, err = eth2Cl.BeaconBlockProposal(ctx, slot, randao, nil)
```

Comment:
not sure this is required, why support eth2client, it will just error? I think just using the workaround is fine.

Reply:
in this case it is not eth2client, it is our beaconmock which implements ethclient's interface and will return our desired block. used this in validatormock_test

Reply:
here i am assuming that if we got empty `addr` parameter then we would assume that we are using beaconmock for testing and not validatorapi.

Reply:
I find also difficult to understand what is happening here, it seems to assume which is the implementation of eth2Cl, if we pass an interface we cannot assume that it is an implemention or another